### PR TITLE
Added a psm action to oms

### PR DESCRIPTION
**Pull request by tculp**

### CODE REVIEW

File: plugins/omar-oms-plugin/grails-app/controllers/omar/oms/ChipperController.groovy [hunk 1]
```diff
@@ -315,51 +275,160 @@ class ChipperController {
...
+            @ApiImplicitParam(name = 'outputFormat', value = 'Output format', defaultValue = '',  allowableValues= "image/jpeg,image/png,image/gif,image/tiff", paramType = 'query', dataType = 'string', required=false),
+            @ApiImplicitParam(name = 'keepBands', value = 'Determine if we auto adjust bands or not', defaultValue = "false", paramType = 'query', dataType = 'boolean', required=false),
+            @ApiImplicitParam(name = 'padThumbnail', value = 'Add padding to the output to make it square', defaultValue = "false", paramType = 'query', dataType = 'boolean', required=false),
+            @ApiImplicitParam(name = 'transparent', value = 'Enable transparent if the outputFormat supports it', defaultValue = "true", paramType = 'query', dataType = 'boolean', required=false),
+    ])
+    def psm(){
+        runChipperCommand("psm")
+    }
+
+    def runChipperCommand(operation) {
```
Comment (Spectrality) [inline plugins/omar-oms-plugin/grails-app/controllers/omar/oms/ChipperController.groovy]: good job refactoring calls with ChipperCommand into one method

File: plugins/omar-oms-plugin/grails-app/controllers/omar/oms/ChipperController.groovy [hunk 2]
```diff
@@ -315,51 +275,160 @@ class ChipperController {
...
+            @ApiImplicitParam(name = 'resamplerFilter', value = 'Which resampling engine to use', defaultValue = '',  allowableValues= "nearest-neighbor, bilinear, cubic, gaussian, blackman, bspline, hanning, hamming, hermite, mitchell, quadratic, sinc, magic", paramType = 'query', dataType = 'string', required=false),
+            @ApiImplicitParam(name = 'outputFormat', value = 'Output format', defaultValue = '',  allowableValues= "image/jpeg,image/png,image/gif,image/tiff", paramType = 'query', dataType = 'string', required=false),
+            @ApiImplicitParam(name = 'keepBands', value = 'Determine if we auto adjust bands or not', defaultValue = "false", paramType = 'query', dataType = 'boolean', required=false),
+            @ApiImplicitParam(name = 'padThumbnail', value = 'Add padding to the output to make it square', defaultValue = "false", paramType = 'query', dataType = 'boolean', required=false),
+            @ApiImplicitParam(name = 'transparent', value = 'Enable transparent if the outputFormat supports it', defaultValue = "true", paramType = 'query', dataType = 'boolean', required=false),
+    ])
+    def psm(){
```
Comment (Spectrality) [inline plugins/omar-oms-plugin/grails-app/controllers/omar/oms/ChipperController.groovy]: does there need to be any sort of input validation on a call to psm? have you tested whether any weird crashes or anything happen when you don't give it two valid images?

File: plugins/omar-oms-plugin/grails-app/controllers/omar/oms/ChipperController.groovy (hunk 1 above)
Comment (tculp) [reply plugins/omar-oms-plugin/grails-app/controllers/omar/oms/ChipperController.groovy]: Thanks

File: plugins/omar-oms-plugin/grails-app/controllers/omar/oms/ChipperController.groovy (hunk 2 above)
Comment (tculp) [reply plugins/omar-oms-plugin/grails-app/controllers/omar/oms/ChipperController.groovy]: We have decided to trust ossim to handle such errors

**Pull request merged**